packages/cozy-stack-client/src/mangoIndex.spec.js

```diff
@@ -237,3 +238,126 @@ describe('getIndexNameFromFields', () => {
     )
   })
 })
+
+describe('makeOperatorsExplicit', () => {
+  it('Transforms implicit $eq operator to explicit', () => {
+    const query = { name: 'test' }
+    const expected = { name: { $eq: 'test' } }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Transforms implicit $and operator to explicit', () => {
+    const query = { name: 'test', age: 42 }
+    const expected = { $and: [{ name: { $eq: 'test' } }, { age: { $eq: 42 } }] }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Maintains explicit $eq operator', () => {
+    const query = { name: { $eq: 'test' } }
+    const expected = { name: { $eq: 'test' } }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Maintains explicit $and operator', () => {
+    const query = { $and: [{ name: 'test' }, { age: 42 }] }
+    const expected = { $and: [{ name: { $eq: 'test' } }, { age: { $eq: 42 } }] }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Handles nested implicit $eq operators', () => {
+    const query = { user: { name: 'test', age: 42 } }
+    const expected = {
+      user: { $and: [{ name: { $eq: 'test' } }, { age: { $eq: 42 } }] }
+    }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Handles nested explicit operators', () => {
+    const query = { user: { $or: [{ name: 'test' }, { age: { $ne: 42 } }] } }
+    const expected = {
+      user: { $or: [{ name: { $eq: 'test' } }, { age: { $ne: 42 } }] }
+    }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Handles mixed implicit and explicit operators', () => {
+    const query = { name: 'test', age: { $ne: 42 } }
+    const expected = { $and: [{ name: { $eq: 'test' } }, { age: { $ne: 42 } }] }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Handles operator with string array', () => {
+    const query = {
+      _id: {
+        $nin: ['id123', 'id456']
+      },
+      type: 'file'
+    }
+    const expected = {
+      $and: [
+        {
+          _id: {
+            $nin: ['id123', 'id456']
+          }
+        },
+        { type: { $eq: 'file' } }
+      ]
+    }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
+  })
+
+  it('Handles operator with string array', () => {
+    const query = {
+      type: {
+        $or: ['konnector', 'worker']
+      }
+    }
+    const expected = {
+      type: {
+        $or: ['konnector', 'worker']
+      }
+    }
+    expect(makeOperatorsExplicit(query)).toEqual(expected)
```

Comment:
Have you tested this one on couchdb? I don't think that would work like this, I would expect this from couchdb: 
```
  "$or": [
    {
      "type": {
        "$eq": "konnector"
      }
    },
    {
      "type": {
        "$eq": "worker"
      }
    }
  ]
  ```

Reply:
Well found it was a failure. I've corrected the function to take this case into account